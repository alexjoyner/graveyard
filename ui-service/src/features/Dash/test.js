import React from 'react';
import { shallow } from 'enzyme';
import { Dash } from '.';

describe('`Dash`', () => {
    let component;
    beforeEach(() => {
        const identity = el => el;
        const props = {
            connectDropTarget: identity
        };
        component = shallow(<Dash {...props}>Testing dash component</Dash>);
    });
    it('renders without exploding', () => {
        expect(component).toMatchSnapshot();
    });
    it('renders children content', () => {
       expect(component.contains('Testing dash component')).toBe(true);
    });
    it('has a class of dash-wrapper', () => {
        expect(component.find('.dash-wrapper').length).toEqual(1);
    });
});