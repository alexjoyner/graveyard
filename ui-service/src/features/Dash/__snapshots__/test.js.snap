// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Dash\` renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dash
    connectDropTarget={[Function]}
>
    Testing dash component
</Dash>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Testing dash component",
      "className": "dash-wrapper",
    },
    "ref": null,
    "rendered": "Testing dash component",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Testing dash component",
        "className": "dash-wrapper",
      },
      "ref": null,
      "rendered": "Testing dash component",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
